apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -1117,6 +1120,26 @@ impl Telemetry {
         TextMapCompositePropagator::new(propagators)
     }
 
+    pub(crate) fn validate_propagation_compatibility(config: &config::Conf) -> Result<(), BoxError> {
+        let propagation = &config.exporters.tracing.propagation;
+        let tracing = &config.exporters.tracing;
+
+        // Check if both Datadog and trace context propagation are active
+        let datadog_active = propagation.datadog || tracing.datadog.enabled();
```

Comment:
1: Is it in any way true that `datadog.enabled()` by itself will turn on `datadog` propagation?

Reply:
Yes, though it might be nice to not duplicate this condition, which is also used in `Telemetry::create_propagator`. Or put a comment on here and inside `create_propagator` that we must keep them in sync